sql: copy hidden columns in CREATE TABLE LIKE

otan opened the pull request:
Resolves #67196

Release note (sql change): CREATE TABLE LIKE now copies hidden columns
over.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/67799)
<!-- Reviewable:end -->

otan commented on pkg/sql/logictest/testdata/logic_test/create_table:
```diff
@@ -340,13 +341,28 @@ SHOW CREATE TABLE like_hash
 ----
 like_hash  CREATE TABLE public.like_hash (
            a INT8 NULL,
-           crdb_internal_a_shard_4 INT4 NOT VISIBLE NOT NULL AS (mod(fnv32(COALESCE(CAST(a AS STRING), '':::STRING)), 4:::INT8)) STORED,
```
so this is because `INCLUDING DEFAULTS` isn't in there, wonder if that's passable.

otan committed:
sql: copy hidden columns in CREATE TABLE LIKE

Also contains logic that restore DEFAULTs for primary key being rowid,
crdb_internal_region and hash sharded indexes.

Release note (sql change): CREATE TABLE LIKE now copies hidden columns
over.

rafiss reviewed (approved):
ty!

otan commented:
bors r=rafiss

mgartner commented:
Thanks!
